components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -773,7 +773,9 @@ private void storeSessionMetaData(String sessionId, HttpServletRequest request)
         String time = Long.toString(System.currentTimeMillis());
 
         Map<String, String> metaDataMap = new HashMap<>();
-        metaDataMap.put(SessionMgtConstants.USER_AGENT, userAgent);
+        if (userAgent != null && !userAgent.isEmpty()) {
```

Comment:
Shall we use StringUtils.isNotEmpty?

Reply:
Resolved with edf57a8684138fdbaf920a416854e5b355231637